### ISSUE TITLE
Request: Format output as JSON

### ISSUE THREAD

**Issue by adamellsworth** [ajoslin/conventional-changelog#35]
It'd be really nice to have the ability to get JSON-formatted Changelogs. I realize the standard is .md, but having JSON output would help immensely in creating a view pertaining to all changes that end-users and binding the .json output as the data source and never have to worry about updating the view itself once the logic is in place.

**Comment by brunobatista** [comment]
+1

**Comment by stevemao** [comment]
Closing for a more generic solution #22

**Comment by stevemao** [comment]
Please checkout the rewrite on the master branch and see if the problem still exists. Thanks.

**Comment by popstas** [comment]
Could you explain how to get json output? I'm tried to get `context` object before generate in writer, but cannot find solution. Should I generate json with handlebars?

**Comment by popstas** [comment]
Only hacky way I found:
``` js
const versions = [];
const stream = conventionalChangelog({
  preset: 'angular',
  releaseCount: 0,
  debug: string => {
    if (!string.includes('Your final context is')) return;
    const version = JSON.parse(string.replace('Your final context is:\n', ''));
    versions.push(version);
  }
});
```

But I don't want to use it.

**Comment by wardellbagby** [comment]
I've got a slightly better way that still feels pretty hacky.

We use the Handlebars templates to just immediately print out the "root" (which is the finalized context with all of your commit data in it). Then we have a little bit of extra work to convert the stream into a string. Every "chunk" from the stream is the JSON version of the root for a release, so we parse that JSON back into objects, combine them together into an array, then print out that resulting array. That gives us an array of "roots" for the releases we requested.

We need to use `json-stringify-safe` as those roots can contain circular references, so JSON.stringify by itself will fail to work. `json-stingify-safe` is what `conventional-changelog-writer` uses under the hood already.

In the example below, I request 2 as requesting 1 will no-op if you're already on the version you're trying to generate a changelog entry for.

```typescript
import Handlebars from 'handlebars';
import stringify from 'json-stringify-safe';
import standardChangelog from 'standard-changelog';

Handlebars.registerHelper(
  'toJSON',
  (data) => new Handlebars.SafeString(stringify(data))
);

const stream = standardChangelog(
  {
    releaseCount: 2,
  },
  null,
  null,
  null,
  {
    mainTemplate: '{{toJSON @root}}',
  }
);

(async () => {
  const chunks = [];

  for await (const chunk of stream) {
    chunks.push(Buffer.from(chunk));
  }

  console.log(
    JSON.stringify(
      Buffer.concat(chunks)
        .toString('utf-8')
        .split('\n')
        .map((value) => JSON.parse(value))
    )
  );
})();
```
